test/unit/v1.0.0/converter-v1-to-v2.test.js

```diff
@@ -698,6 +698,150 @@ describe('v1.0.0 to v2.0.0', function () {
             });
         });
 
+        it('should set mode to raw if dataMode is not set but all other data props are set', function (done) {
```

Comment:
Also, need tests to check for the entire precedence order working correctly.

Reply:
Added test for `formdata` as well.